eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -281,9 +281,43 @@ jobs:
           jdkArchitectureOption: 'x64'
           publishJUnitResults: false
           goals: ${{ parameters.TestGoals }}
-        # we want to run this when TestFromSource isn't true (which covers normal running when it isn't set)
-        # OR when ShouldRunSourceTests is true
-        condition: and(succeeded(), or(ne(variables['TestFromSource'],'true'), eq(variables['ShouldRunSourceTests'],'true')))
+        # we want to run this when TestFromSource isn't true
+        condition: and(succeeded(), ne(variables['TestFromSource'],'true'))
+
+      - task: Maven@3
+        displayName: 'Build for From Source run'
+        inputs:
+          mavenPomFile: pom.xml
+          goals: 'install'
+          # Build everything in the root pom.xml and install it. The purpose here is to build and install everything after
+          # the version have been manipulated. The purpose is to have all of the libraries installed on the machine so we
+          # can simply run tests with the -amd switch and it won't have to deal with the AdditionalModules snafu that's causing
+          # us to list every transitive azure client dependency.
+          # Note: yes, this overbuilds, it builds data track libraries but that'll be looked at later.
+          options: '$(DefaultOptions) -DskipTests -Dgpg.skip -Dmaven.javadoc.skip=true -Dcheckstyle.skip=true -Dspotbugs.skip=true -Drevapi.skip=true'
```

Comment:
May be able to speed this up by using `-T 1C` to allow Maven to parallelize install.

Reply:
Yup, that's the plan with the next commit.